src/sentry/models/release.py

```diff
@@ -381,61 +381,70 @@ def set_commits(self, commit_list):
+                                email=author_email)
+                        authors[author_email] = author
                     else:
                         author = authors[author_email]
 
-                    defaults = {
+                    commit_data = {
                         'message': data.get('message'),
-                        'author': author,
                         'date_added': data.get('timestamp') or timezone.now(),
                     }
-                    commit, created = Commit.objects.get_or_create(
+                    # If we didn't get an author don't overwrite an existing one.
+                    if author is not None:
+                        commit_data['author'] = author
+
+                    commit, created = Commit.objects.create_or_update(
                         organization_id=self.organization_id,
                         repository_id=repo.id,
                         key=data['id'],
-                        defaults=defaults,
-                    )
+                        defaults=commit_data,
+                        values=commit_data)
+                    if not created:
+                        commit = Commit.objects.get(
+                            organization_id=self.organization_id,
+                            repository_id=repo.id,
+                            key=data['id'])
+
                     if author is None:
                         author = commit.author
 
                     commit_author_by_commit[commit.id] = author
 
                     patch_set = data.get('patch_set', [])
-
                     for patched_file in patch_set:
-                        CommitFileChange.objects.get_or_create(
+                        commit_file_data = {
+                            'filename': patched_file['path'],
+                            'type': patched_file['type'],
+                        }
+                        CommitFileChange.objects.create_or_update(
                             organization_id=self.organization.id,
                             commit=commit,
-                            filename=patched_file['path'],
-                            type=patched_file['type'],
+                            defaults=commit_file_data,
+                            values=commit_file_data
                         )
 
-                    if not created:
-                        update_kwargs = {}
-                        if commit.message is None and defaults['message'] is not None:
-                            update_kwargs['message'] = defaults['message']
-                        if commit.author_id is None and defaults['author'] is not None:
-                            update_kwargs['author'] = defaults['author']
-                        if defaults.get('date_added') is not None:
-                            update_kwargs['date_added'] = defaults['date_added']
-                        if update_kwargs:
-                            commit.update(**update_kwargs)
```

Comment:
This block was emulating the behavior that `create_or_update` has as it pulls from `defaults` defined on line 396 to make an update, which is why I've removed this logic.

Reply:
It looks like this logic is intended to not overwrite existing values for message/author_id if they already exist. This diff changes that, I'm not sure what the effect of this change would be.

Reply:
I think not overwriting is a mistake/error though. If an API request comes in to update a set of sub-resources and a significant part of those changes is ignored my first reaction would be that the API is broken.